eZ/Publish/Core/Persistence/Legacy/Content/UrlAlias/Handler.php

```diff
@@ -957,12 +957,18 @@ protected function getRealAliasId($locationId)
      * @param array $actionMap
      * @param mixed $oldParentAliasId
      * @param mixed $newParentAliasId
+     * @param string[] $alreadyGeneratedPairs
      */
-    protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId)
+    protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId, &$alreadyGeneratedPairs = [])
```

Comment:
I'm generally against using arrays passed as a reference as argument. In this case it makes sense purely because storing it in property might be cumbersome. We can't really return the value here, as that would be a BC break.

Sooooooo... :+1:

```suggestion
    protected function copySubtree($actionMap, $oldParentAliasId, $newParentAliasId, array &$alreadyGeneratedPairs = [])
```

Reply:
> I'm generally against using arrays passed as a reference as argument. In this case it makes sense purely because storing it in property might be cumbersome. We can't really return the value here, as that would be a BC break.
> 
> Sooooooo... +1

There's no BC promise for a protected method inside `eZ/Publish/Core/` namespace. It's all there stated in the [Readme.md](https://github.com/ezsystems/ezpublish-kernel/blob/7.5/Readme.md#current-organization) ;-)